src/battle_anim_ice.c

```diff
@@ -1342,71 +1354,86 @@ void AnimTask_Hail(u8 taskId)
 static void AnimTask_Hail2(u8 taskId)
 {
     struct Task *task = &gTasks[taskId];
-    switch (task->data[0])
+    switch (task->tState)
     {
     case 0:
-        if (++task->data[4] > 2)
+        if (++task->tInitialDelayTimer > 2)
         {
-            task->data[4] = 0;
-            task->data[5] = 0;
-            task->data[2] = 0;
-            task->data[0]++;
+            task->tInitialDelayTimer = 0;
+            task->tHailSpawnTimer = 0;
+            task->tHailAffineAnimNum = 0;
+            task->tState++;
         }
         break;
     case 1:
-        if (task->data[5] == 0)
+        if (task->tHailSpawnTimer == 0)
         {
-            if (GenerateHailParticle(task->data[3], task->data[2], taskId, 1))
-                task->data[1]++;
+            if (GenerateHailParticle(task->tHailStructId, task->tHailAffineAnimNum, taskId, 1))
+                task->tSpriteCount++;
 
-            if (++task->data[2] == 3)
+            if (++task->tHailAffineAnimNum == 3) // i.e. ARRAY_COUNT(sAffineAnims_HailParticle)
             {
-                if (++task->data[3] == 10)
-                    task->data[0]++;
+                if (++task->tHailStructId == 10) // i.e. ARRAY_COUNT(sHailCoordData)
```

Comment:
I'm assuming these are listed as comments because they didn't match? You can usually fix that by casting the result of `ARRAY_COUNT` to `int`